### ISSUE TITLE
Option to disable static libraries

### ISSUE THREAD

**Issue by rossburton** [openssl/openssl#8823]
There's a `no-shared` option to disable the building of shared libraries. Would it be possible to add a `no-static` option to disable the building of static libraries?

**Comment by levitte** [comment]
That's going to be unpractical for a number of reasons...  but let's take it from a different angle: is this about what gets *installed*, i.e. the result of `make install`?

**Comment by rossburton** [comment]
On Sat, 11 May 2019 at 22:51, Richard Levitte <notifications@github.com>
wrote:

> That's going to be unpractical for a number of reasons... but let's take
> it from a different angle: is this about what gets *installed*, i.e. the
> result of make install?
>

Yes.  Obviously a workaround is to just delete those manually.

Ross

**Comment by levitte** [comment]
Not quite, because the installation will fail if files are unexpectedly missing...  But ok, means I caught your issue right.  This isn't a configuration request, it's an installation option request.

I'm not sure if this is high priority enough to actually happen, but I won't close this immediately.

**Comment by nhorman** [comment]
marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.

**Comment by emmenlau** [comment]
I would have been interested in this too.

Motivation: Having an explicit option to not install the static libraries makes sure that my build does not accidentally link the static versions. I'm building most code as shared, and linking the static libraries (accidentally) would add the binary into every shared library, with potentially desastrous results.

Being unable to avoid the static libraries puts quite some burden on myself, because tools like cmake will "just magically" detect and link openssl for me, and I have not sufficiently rigid control over what exactly was linked.

Would it be possible to re-consider this issue?
